gips/data/landsat/landsat.py

```diff
@@ -1346,3 +1393,200 @@ def _readraw(self, asset_type):
 
         verbose_out('%s: read in %s' % (image.Basename(), datetime.now() - start), 2)
         return image
+
+    def sentinel2_coreg_export(self, tmpdir, utm_zone):
+        landsat_shp = landsatRepository.get_setting('tiles')
+        sentinel2Data = utils.import_data_class('sentinel2')
+        sentinel2Asset = sentinel2Data.Asset
+        spatial_extent = SpatialExtent.factory(sentinel2Data, site=landsat_shp, where="pr = '%s'" % self.id, pcov=33.0)[0]
+        temporal_extent = TemporalExtent(self.date.strftime("%Y-%j"))
+
+        self._time_report("querying for most recent sentinel2 images")
+        inventory = DataInventory(sentinel2Data, spatial_extent, temporal_extent, fetch=False)
+
+        # If there is no available sentinel2 scene on that day, search before and after
+        # until one is found.
+        delta = timedelta(1)
+        date_found = self.date
+        while len(inventory) == 0:
+            temporal_extent = TemporalExtent((self.date + delta).strftime("%Y-%j"))
+            inventory = DataInventory(sentinel2Data, spatial_extent, temporal_extent, fetch=False)
+
+            if len(inventory) != 0:
+                date_found = self.date + delta
+                break
+
+            temporal_extent = TemporalExtent((self.date - delta).strftime("%Y-%j"))
+            inventory = DataInventory(sentinel2Data, spatial_extent, temporal_extent, fetch=False)
+
+            date_found = self.date - delta
+            delta = delta + timedelta(1)
+
+        geo_images = []
+        tiles = inventory.data.values()[0].tiles.keys()
+        for tile in tiles:
+            assets = sentinel2Asset.discover(tile, date_found, 'L1C')
+            band_8 = [f for f in assets[0].datafiles() if f.endswith('B08.jp2')]
+            assets[0].extract(band_8, tmpdir)
+            geo_images.append(os.path.join(tmpdir, band_8[0]))
+
+        self._time_report("merge sentinel images to bin")
+        merge_args = ["gdal_merge.py", "-o", tmpdir + "/sentinel_mosaic.bin", "-of", "ENVI", "-a_nodata", "0"]
+        # only use images that are in the same proj as landsat tile
+        merge_args.extend([i for i in geo_images if basename(i)[1:3] == utm_zone])
+        subprocess.call(merge_args)
+
+        self._time_report("done with s2 export")
+        return tmpdir + '/sentinel_mosaic.bin'
+
+    def run_arop(self, base_band_filename, utm_zone):
```

Comment:
Optional:  The inner functions below seem to be used only once, mostly, so I wonder if it'd be better to just run the code inline and save the indirection and verbosity.